### PR TITLE
Update kubernetes_metadata to 0.21, add new flags

### PR DESCRIPTION
**Pull request by therc**
Add ELASTICSEARCH_DYNAMIC, FLUENTD_LOGSTASH_PREFIX, KUBERNETES_PRESERVE_JSON_LOG
Tweak description of FLUENTD_LOGSTASH_FORMAT

### CODE REVIEW

**Comment by jimmidyson** [comment]
LGTM thanks!

File: start-fluentd [hunk 1]
```diff
@@ -32,16 +34,17 @@ cat << 'EOF' > /etc/fluent/fluent.conf
 
 <filter kubernetes.**>
   type kubernetes_metadata
+  preserve_json_log ${KUBERNETES_PRESERVE_JSON_LOG}
```
Comment (jimmidyson) [inline start-fluentd:37]: Oops just noticed this needs moving in to the other heredoc - fixing up in separate commit.

**Comment by therc** [comment]
Thanks. Builds are failing when installing g++:

https://hub.docker.com/r/fabric8/fluentd-kubernetes/builds/bgyhgkitxdgpthjc7mjuw6w/

**Comment by jimmidyson** [comment]
Seems to be a bug in building Docker images - perhaps related to https://github.com/docker/docker/issues/10180 & other bugs.

Trying to work around it.

**Comment by jimmidyson** [comment]
Phew fixed - `:v1.13` will be available soon (or you can use `:latest` with the normal provisos).

**Comment by therc** [comment]
It worked! Thank you so much. My logs should be at least 33% smaller now. :-)

**Comment by jimmidyson** [comment]
Thank you for sending in the PR!

**Pull request merged**